proposal/color-4-new-spaces.md

```diff
@@ -0,0 +1,1790 @@
+
+      * Let `normal-channel` be the result of converting `channel` to `deg`
+        allowing unitless.
+
+      * Append `normal-channel` as the next item in `normal`.
+
+      > Normalizing the result into a half-open range of `[0,360)` would be a
+      > lossy transformation, since some forms of [hue interpolation][hue-method]
+      > require the specified hue values.
+
+    * Otherwise, if `valid` requires a percentage:
+
+      * If `channel` is a number with units other than `%`, throw an error.
+
+      * Append `channel` as the next item in `normal`.
+
+    * Otherwise:
+
+      * Set `channel` to the result of [percent-converting] `channel` with a
+        `max` defined by the `valid` channel range.
+
+      * Append `channel` as the next item in `normal`.
+
+* Return `normal`.
+
+### Interpolating Colors
+
+> This procedure is based on the
+> [color interpolation](https://drafts.csswg.org/css-color-4/#interpolation)
+> procedures defined in [CSS Color Level 4][color-4].
+
+This procedure accepts two color arguments (`color1` and `color2`), a
+[color interpolation method] `method`, and a percentage `weight` for `color1`
+in the mix. It returns a new color `mix` that represents the appropriate mix of
+input colors.
+
+* If either `color1` or `color2` is not a color, throw an error.
+
+* If `weight` is null, set `weight` to `50%`.
+
+* Set `weight` to the result of [percent-converting] `weight` with a max of 1,
+  and then clamping the value between 0 and 1, inclusive.
+
+* Otherwise:
+
+  * If `method` is not a [color interpolation method][color-method], throw an
+    error.
+
+  * Let `space` be the _interpolation color space_ specified in `method`.
+
+  * If `space` is a [PolarColorSpace][color-method]:
+
+    * Let `hue-arc` be the `HueInterpolationMethod` specified in `method`, or
+      `shorter` if no hue interpolation is specified.
+
+* For each `color` of `color1` and `color2`:
+
+  * Let `origin-space` be `color`'s color space.
+
+  * If `origin-space` is not a [known color space], throw an error.
+
+  * Let `missing` be a list of channel names in `color` that are [missing].
+
+  * Set `color` to the results of [converting] `color` into `space`.
+
+  * For each `channel` in `missing`:
+
+    * If `space` has an [analogous component][missing] to `channel`, set
+      `channel` to `none`.
+
+  * If any `component` of `color` is `none`, set that `component` to the value
+    of the corresponding component in the other color.
+
+    > If both values are `none`, the interpolation result for that component
+    > will also be `none`.
+
+  * Set `color` to the result of [premultiplying] `color`.
+
+* Let `mix` be a new color in the [known color space] `space`, with `none` for
+  alpha and all channel values.
+
+* For each `channel` of `mix`:
+
+  * Let `channel1` and `channel2` be the corresponding channel values in
+    `color1` and `color2` respectively.
+
+  * If `channel` represents a hue angle, set `channel1` and `channel2`
```

Comment:
I'd say "If `channel` has a polar angle value" to match the definition in Known Color Space.